subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1290,6 +1299,119 @@ static void le_start_encryption(struct net_buf *buf, struct net_buf **evt)
 	*evt = cmd_status(status);
 }
 #endif /* CONFIG_BT_CTLR_LE_ENC */
+
+#if defined(CONFIG_BT_CTLR_CENTRAL_ISO)
+static void le_set_cig_parameters(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_set_cig_params *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_set_cig_params *rp;
+	uint32_t m_interval;
+	uint32_t s_interval;
+	uint16_t m_latency;
+	uint16_t s_latency;
+	uint8_t status;
+	uint8_t i;
+
+	m_interval = sys_get_le24(cmd->m_interval);
+	s_interval = sys_get_le24(cmd->s_interval);
+	m_latency = sys_le16_to_cpu(cmd->m_latency);
+	s_latency = sys_le16_to_cpu(cmd->s_latency);
+
+	/* Create CIG or start modifying existing CIG */
+	status = ll_cig_parameters_open(cmd->cig_id, m_interval, s_interval,
+					cmd->sca, cmd->packing, cmd->framing,
+					m_latency, s_latency, cmd->num_cis);
+
+	rp = hci_cmd_complete(evt, sizeof(*rp) +
+				   cmd->num_cis * sizeof(uint16_t));
+	rp->cig_id = cmd->cig_id;
+	rp->num_handles = cmd->num_cis;
+
+	/* Configure individual CISes */
+	for (i = 0; !status && i < cmd->num_cis; i++) {
+		struct bt_hci_cis_params *params = cmd->cis;
+		uint16_t handle;
+		uint16_t m_sdu;
+		uint16_t s_sdu;
+
+		m_sdu = sys_le16_to_cpu(params->m_sdu);
+		s_sdu = sys_le16_to_cpu(params->s_sdu);
+
+		status = ll_cis_parameters_set(params->cis_id, m_sdu, s_sdu,
+					       params->m_phy, params->s_phy,
+					       params->m_rtn, params->s_rtn,
+					       &handle);
+		rp->handle[i] = sys_cpu_to_le16(handle);
+	}
+
+	/* Only apply parameters if all went well */
+	if (!status) {
+		status = ll_cig_parameters_commit(cmd->cig_id);
+	}
+
+	rp->status = status;
+}
+
+static void le_create_cis(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_create_cis *cmd = (void *)buf->data;
+	uint8_t status;
+	uint8_t i;
+
+	/*
+	 * Creating new CISes is disallowed until all previous CIS
+	 * established events have been generated
+	 */
+	if (cis_pending_count) {
+		*evt = cmd_status(BT_HCI_ERR_CMD_DISALLOWED);
+		return;
+	}
+
+	/* Check all handles before actually starting to create CISes */
+	status = 0x00;
+	for (i = 0; !status && i < cmd->num_cis; i++) {
+		uint16_t cis_handle;
+		uint16_t acl_handle;
+
+		cis_handle = sys_le16_to_cpu(cmd->cis[i].cis_handle);
+		acl_handle = sys_le16_to_cpu(cmd->cis[i].acl_handle);
+		status = ll_cis_create_check(cis_handle, acl_handle);
```

Comment:
Is the idea that `ll_cis_create_check` will also check for duplicate CIS handles?

Reply:
Interestingly, the spec is quiet about what should happen in the case of duplicate CIS handles, but I will try to clarify.

Reply:
I guess it could be argued that if there's a duplicate CIS handle, it falls under 

> If the Host attempts to create a CIS that is already created, the Controller shall reject the command and return the error code Connection Already Exists (0x0B).

Reply:
Erratum 16058 has been created to clarify what exactly is supposed to happen if there are duplicate CIS handles.